test/e2e/lib/jest/globalSetup.ts

```diff
@@ -0,0 +1,65 @@
+import { mkdir } from 'fs/promises';
+import path from 'path';
+import config from 'config';
+import { chromium } from 'playwright';
+
+// Save cookies for defaultUser.
+export default async (): Promise< void > => {
```

Comment:
Alright, I want to preface this feedback as that I realize it's asking a lot for potentially a small change, so we don't definitely need to do this, but I'll make my pitch 🙂 

The short is: it feels weird for me to not have the bulk of this setup function be in the `calypso-e2e` package. What I had imagined was a function in `calypso-e2e` with a signature like...

```typescript
cacheLoginCookiesForAccounts( accounts: string[], cookieBasePath: string): Promise<void>
```

And then this global setup function would call that function and pass in the account names and the path to use for the cookies.

As I see it, there a couple drawbacks from not having the core functionality in `calypso-e2e`:
1. We're having to re-implement some functionality (like logging in, spinning up a browser context, etc) that is already implemented in the `calypso-e2e` package. It would be best if we could re-use what we've already implemented!
2. We're forced to split implementation details across two packages, and this ends up breaking encapsulation. For example, it feels weird in this file that we have to know the specific cookie ENV variable name we need to set for the `calypso-e2e` package to use later, and to know the naming scheme we need to use for saving off cookies (to later be consumed again by a function in the `calypso-e2e` package). I think that might lead to fragility over time, and it would make it harder for future consumers of the `calypso-e2e` package -- they effectively have to read the underlying code to use the library.

But, as you mention, the big blocker here is that `calypso-e2e` uses `@jest/globals`, which got me thinking... Why does it?

I think it would be powerful if we could completely decouple the `calypso-e2e` package from `jest` - this means that anyone could use this project and then use whatever test runner they want! It also means that if we want to switch runners in the future (to the Playwright runner, e.g.), we wouldn't have to touch the `calypso-e2e` package at all, making the change even easier!

To do that decoupling, we only have to do a couple relatively small things:
1. Move the `jest-conditionals.ts` file to `test/e2e/lib` - it's coupled to jest, so I think it makes sense to live with the rest of jest stuff
2. Change `setupHooks` so that it follows more of a dependency injection pattern. There's a couple different ways we could do this: we could have `setupHooks` accept the before/after functions. Or alternatively, we could have `calypso-e2e` just export functions that do the bulk of what is called during before/after, and then we could have a local `setupHooks` function in `test/e2e/lib` that calls those functions within the jest `beforeAll` and `afterAll` functions.


Anyway, I realize that's a lot, and maybe way too much for here, but wanted to throw the idea and discussion out there!